nixos/modules/services/networking/ddclient.nix

```diff
@@ -262,6 +262,17 @@ in
       lib.optional (cfg.use != "")
         "Setting `use` is deprecated, ddclient now supports `usev4` and `usev6` for separate IPv4/IPv6 configuration.";
 
+    assertions = [
+      {
+        assertion = !((cfg.passwordFile != null) && (cfg.secretsFile != null));
+        message = "You cannot use both passwordFile and secretsFile at the same time.";
```

Comment:
I don't think these assertion messages have any context when presented to the user, so I'd personally change e.g. "passwordFile" to "services.ddclient.passwordFile".

I plan on approving the PR, but wait a few days with merging. So if you agree about being explicit about the option names, just force push in the meantime :-)